### PR TITLE
Resolving Roslyn Analyzer Issues

### LINKED ISSUE #5005: Resolve Roslyn Analyzer issues
Internal issue: 776514/

### PR DESCRIPTION
**Pull request by RaymonGulati1**
Fix #5005 

Fix BuildTask to include assembly information such as title, name, version. 
Update strings to compare with culturally invariant process

### CODE REVIEW

File: Common/Tools/BuildTasks/Properties/AssemblyInfo.cs [hunk 1]
```diff
@@ -0,0 +1,33 @@
+﻿using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+// General Information about an assembly is controlled through the following 
+// set of attributes. Change these attribute values to modify the information
+// associated with an assembly.
+[assembly: AssemblyTitle("Microsoft.VisualStudioTools.BuildTasks.Properties")]
```
Comment (huguesv) [inline Common/Tools/BuildTasks/Properties/AssemblyInfo.cs]: Remove `.Properties` from title and product

File: Python/Product/Cookiecutter/Attributes.cs [hunk 2]
```diff
@@ -26,7 +27,7 @@ internal sealed class SRDisplayNameAttribute : DisplayNameAttribute {
             _name = name;
         }
 
-        public override string DisplayName => Strings.ResourceManager.GetString(_name);
+        public override string DisplayName => Strings.ResourceManager.GetString(_name, CultureInfo.InvariantCulture);
```
Comment (huguesv) [inline Python/Product/Cookiecutter/Attributes.cs]: We don't want to pass `InvariantCulture` here.

When using the generated code in Strings.Designer.cs, you can see that all calls are like this one:
```
ResourceManager.GetString("SearchPage_LinksGitHubIssues", resourceCulture);
```

`resourceCulture` is private but there's an accessor `Culture`.

So the code should be:
```
Strings.ResourceManager.GetString(_name, Strings.Culture)
```

Comment (huguesv) [reply Python/Product/Cookiecutter/Attributes.cs]: Same for all the other `ResourceManager.GetString` calls.

File: Python/Product/VSInterpreters/PackageManager/PipRequirementsUtils.cs [hunk 3]
```diff
@@ -91,7 +91,7 @@ IEnumerable<PackageSpec> installed
                 var installedPackage = installed.FirstOrDefault(pkg =>
                     string.Compare(pkg.Name, requirementPackageName, StringComparison.OrdinalIgnoreCase) == 0);
 
-                if (!requirementPackageName.Equals("") && installedPackage == null) {
+                if (!requirementPackageName.Equals("", StringComparison.InvariantCulture) && installedPackage == null) {
```
Comment (huguesv) [inline Python/Product/VSInterpreters/PackageManager/PipRequirementsUtils.cs]: `if (requirementPackageName.Length > 0 && installedPackage == null)`

File: Common/Tools/BuildTasks/Properties/AssemblyInfo.cs [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+[assembly: AssemblyTrademark("")]
+[assembly: AssemblyCulture("")]
+
+// Setting ComVisible to false makes the types in this assembly not visible 
+// to COM components.  If you need to access a type in this assembly from 
+// COM, set the ComVisible attribute to true on that type.
+[assembly: ComVisible(false)]
+
+// The following GUID is for the ID of the typelib if this project is exposed to COM
+[assembly: Guid("861b9a5e-f07e-436b-a737-24072a2e6b55")]
```
Comment (huguesv) [inline Common/Tools/BuildTasks/Properties/AssemblyInfo.cs]: You can remove the Guid attribute

File: Python/Product/VSInterpreters/PackageManager/PipPackageManager.cs [hunk 5]
```diff
@@ -716,8 +716,10 @@ private sealed class Suppressed : IDisposable {
         }
 
         private void OnChanged(object sender, FileSystemEventArgs e) {
-            if ((Directory.Exists(e.FullPath) && !"__pycache__".Equals(PathUtils.GetFileOrDirectoryName(e.FullPath))) ||
-                ModulePath.IsPythonFile(e.FullPath, false, true, false)) {
+            if ((Directory.Exists(e.FullPath) && 
+                !PathUtils.GetFileOrDirectoryName(e.FullPath).Equals("__pycache__", StringComparison.InvariantCulture)) ||
+                ModulePath.IsPythonFile(e.FullPath, false, true, false)
+            ) {
```
Comment (huguesv) [inline Python/Product/VSInterpreters/PackageManager/PipPackageManager.cs]: Searching for `.Equals(` in the code base shows several examples that compare file names with `StringComparison.OrdinalIgnoreCase`, and I think that's what we should use here.

File: Python/Product/Cookiecutter/Shared/Interpreters/InterpreterConfiguration.cs [hunk 6]
```diff
@@ -98,7 +98,7 @@ public sealed class InterpreterConfiguration {
 
             var cmp = StringComparer.OrdinalIgnoreCase;
             return cmp.Equals(PrefixPath, other.PrefixPath) &&
-                string.Equals(Id, other.Id) &&
+                string.Equals(Id, other.Id, StringComparison.InvariantCultureIgnoreCase) &&
```
Comment (huguesv) [inline Python/Product/Cookiecutter/Shared/Interpreters/InterpreterConfiguration.cs]: I'm hesitant here, but the more I think about it - the `Id` may contain a project name or folder name, so I'm thinking we better use `cmp` for the `Id` as well. Same for the duplicate code below.

File: Common/Tools/BuildTasks/Properties/AssemblyInfo.cs (hunk 1 above)
Comment (RaymonGulati1) [reply Common/Tools/BuildTasks/Properties/AssemblyInfo.cs]: Fixed in next commit

File: Common/Tools/BuildTasks/Properties/AssemblyInfo.cs (hunk 4 above)
Comment (RaymonGulati1) [reply Common/Tools/BuildTasks/Properties/AssemblyInfo.cs]: Why?

File: Python/Product/VSInterpreters/PackageManager/PipRequirementsUtils.cs (hunk 3 above)
Comment (RaymonGulati1) [reply Python/Product/VSInterpreters/PackageManager/PipRequirementsUtils.cs]: Modified in next commit

File: Python/Product/VSInterpreters/PackageManager/PipPackageManager.cs (hunk 5 above)
Comment (RaymonGulati1) [reply Python/Product/VSInterpreters/PackageManager/PipPackageManager.cs]: Will be modified in next commit

File: Python/Product/Cookiecutter/Shared/Interpreters/InterpreterConfiguration.cs (hunk 6 above)
Comment (RaymonGulati1) [reply Python/Product/Cookiecutter/Shared/Interpreters/InterpreterConfiguration.cs]: Will be modified in next commit

File: Python/Product/Cookiecutter/Attributes.cs (hunk 2 above)
Comment (RaymonGulati1) [reply Python/Product/Cookiecutter/Attributes.cs]: Will be modified in next commit

File: Common/Tools/BuildTasks/Properties/AssemblyInfo.cs (hunk 4 above)
Comment (huguesv) [reply Common/Tools/BuildTasks/Properties/AssemblyInfo.cs]: We don't expose anything as a COM type. I know some of our projects have unnecessary Guid attribute, but prefer to not introduce more.

Comment (RaymonGulati1) [reply Common/Tools/BuildTasks/Properties/AssemblyInfo.cs]: Modified in next commit

**Pull request merged**